Example/Tests/TensorIOModelBundleValidatorTests.mm

```diff
@@ -0,0 +1,2174 @@
+    
+    [properties removeObjectForKey:@"id"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutVersionDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"version"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutAuthorDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"author"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutLicenseDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"license"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutModelDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"model"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutInputsDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"inputs"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithoutOutputsDoesNotValidate {
+    // it should not validate
+    
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    [properties removeObjectForKey:@"outputs"];
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertFalse(valid);
+    XCTAssertNotNil(error);
+}
+
+- (void)testBundleWithValidPropertiesValidates {
+    NSError *error;
+    NSMutableDictionary *properties = self.basicProperties.mutableCopy;
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertTrue(valid);
+    XCTAssertNil(error);
+}
+
+// MARK: - Model Properties Validation
+
+- (void)testModelPropertiesValidates {
```

Comment:
Would split this up into different tests.

Reply:
Again, not blocker and it's largely a matter of preference -- just if the tests ever fail, it's more informative if the tests are smaller.